awx/main/isolated/manager.py

```diff
@@ -187,6 +188,13 @@ def check(self, interval=None):
                                                       self.private_data_dir,
                                                       extravars=extravars)
             status, rc = runner_obj.status, runner_obj.rc
+            if self.check_callback is not None:
+                command_path = self.path_to('artifacts', self.ident, 'command')
+                # If the command artifact has been synced back, update the model
+                if os.path.exists(command_path):
+                    with open(command_path, 'r') as f:
+                        data = json.load(f)
+                    self.check_callback(data['command'], data['cwd'], self.runner_params['envvars'].copy())
```

Comment:
Okay, so `runner_params` isn't quite what we want here.  This is the environment variables the *controller* uses when it runs a playbook to sync data *to* the isolated node.